src/run/runner/setup.rs

```diff
@@ -91,7 +91,30 @@ lazy_static! {
     };
 }
 
+fn check_installed_valgrind() -> Result<bool> {
+    let is_valgrind_installed = Command::new("which")
+        .arg("valgrind")
+        .output()
+        .is_ok_and(|output| output.status.success());
+    if !is_valgrind_installed {
+        return Ok(false);
+    }
+
+    let version_output = Command::new("valgrind").arg("--version").output()?;
+    if !version_output.status.success() {
+        return Ok(false);
+    }
```

Comment:
I think this function shouldn't return a Result, you can reuse the same you used before:
```
        .output()
        .is_ok_and(|output| output.status.success());
```